### ISSUE TITLE
Calling UTIL.BlastDamage on a crouching player after death crashes the server

### ISSUE THREAD

**Issue by KingDavidW** [Facepunch/garrysmod-issues#4785]
### Details
If a player is crouching and dies, with util.BlastDamage immediately called at that position, it will crash the server. 

### Steps to reproduce
Here's shared code you can test. Run the command, watch the world burn:

```
if SERVER then
	concommand.Add( "crash_the_server", function( ply, cmd, args )
		ply:SetNWBool( "CRASHTIME", true )
                timer.Simple( 0.3, function() //Buffer to make sure they are crouched
		    ply:Kill()
                end )
	end )

	hook.Add( "PlayerDeath", "crash_hooker", function( ply, hitgroup, dmginfo )
		util.BlastDamage( ply, ply, ply:GetPos(), 50, 100 )
	end )
end

hook.Add( "StartCommand", "wOS.LastStand.PreventAttack", function( ply, cmd )
	if !ply:GetNWBool( "CRASHTIME", false ) then return end
	cmd:SetButtons( bit.bor( cmd:GetButtons(), IN_DUCK ) )
end )
```

### Additional Notes
This isn't exclusive to StartCommand or CreateMove hook generation. This is literally if the player is crouching. You can hold the crouch key yourself after calling the command and crash the server.

This was tested on a Linux and Windows dedicated server, both running the main branch

**Comment by robotboy655** [comment]
You are basically hitting an infinite loop.

**Comment by robotboy655** [comment]
Fixed ability to crash the game when the dying player receives damage while in a death related hook. This results in Player.Alive returning false ( used to return true ) in PlayerDeath and DoPlayerDeath hooks for the dying player.

The same issue affects EntityTakeDamage hooks, they are not as easy to solve and I am not sure what the correct solution would be. I have added warnings to the ones affected.
